public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -180,6 +178,11 @@ export class CloudWatchDatasource
       region: this.replace(this.getActualRegion(target.region), options.scopedVars, true, 'region'),
     }));
 
+    const startTime = new Date();
+    const timeoutFunc = () => {
+      return Date.now() >= startTime.valueOf() + rangeUtil.intervalToMs(this.logsTimeout);
+    };
```

Comment:
this timeoutFunction is already defined in `runWithRetry`, I think that you just need to remove `timeoutFunction` from the `options` parameter

Reply:
They're two separate function/api calls, so creating this timeoutFunc and using it in both places means that they run on the same timeout.